app/src/androidTest/java/com/omen273/crossLingo/FreeClueInstrumentedTest.kt

```diff
@@ -63,13 +64,49 @@ class FreeClueInstrumentedTest : TestBaseClass() {
             menuClick(R.string.solve_square_free, R.id.menu_solve_cell)
         }
 
+        // 4th word
+        menuClick(R.string.solve_word, R.id.menu_solve_word)
+        starNumber -= GameActivity.WORD_OPEN_PRICE
+
         val start3 = System.currentTimeMillis()
         waitForCondition("", { System.currentTimeMillis() - start3 > waitingTime },
             waitingTime + 1)
         menuClick(R.string.solve_square, R.id.menu_solve_cell)
         pressBack()
         waitForView(ViewMatchers.withId(R.id.tableLayout))
-        starNumber -= LETTER_OPEN_PRICE
+        starNumber -= GameActivity.LETTER_OPEN_PRICE
+        waitForCondition("star checking", { starNumber == readStarsFromConfig() })
+
+        loadFirstCrossword()
+
+        // 5th word
+        val start4 = System.currentTimeMillis()
+        waitForCondition("", { System.currentTimeMillis() - start4 > waitingTime },
+            waitingTime + 1)
+        menuClick(R.string.solve_word_free, R.id.menu_solve_word)
+
+        // 8 more words to solve all except one word
+        for (i in 0..7)
+        {
+            menuClick(R.string.solve_word, R.id.menu_solve_word)
+            starNumber -= GameActivity.WORD_OPEN_PRICE
+        }
+
+        val start5 = System.currentTimeMillis()
```

Comment:
Please test also that we have free tip for cell

Reply:
done